### PR TITLE
Config support for fallthrough attribute

### PR DESCRIPTION
**Pull request by mcopik**
Adds a feature test for C++17 `[[fallthrough]]` and removes gcc 7.1 warnings complaining about a possible fallthrough.

### CODE REVIEW

File: hpx/config/attributes.hpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_CONFIG_ATTRIBUTES_HPP
+#define HPX_CONFIG_ATTRIBUTES_HPP
+
+#include <hpx/config/defines.hpp>
+
+#if defined(HPX_HAVE_CXX17_FALLTHROUGH_ATTRIBUTE)
+#   define HPX_FALLTHROUGH [[fallthrough]];
```
Comment (hkaiser) [inline hpx/config/attributes.hpp]: I'd suggest to remove the semicolon from the macro definition. Let's leave this to the user. This is not a big issue, though, some compilers warn about duplicate semicolons, though...

Comment (mcopik) [inline hpx/config/attributes.hpp]: Duplicate semicolon because the user may follow the macro with a semicolon?
`HPX_FALLTHROUGH;`

Comment (hkaiser) [inline hpx/config/attributes.hpp]: Yes, that's what I meant. But the main rationale would be to have the semicolon in user code. A casual reader might not 'know' that `HPX_FALLTHROUGH` is a macro...

Comment (mcopik) [inline hpx/config/attributes.hpp]: I agree, thanks for the review! Fixed.

**Comment by hkaiser** [comment]
FWIW: this describes how things are implemented in gcc: https://developers.redhat.com/blog/2017/03/10/wimplicit-fallthrough-in-gcc-7/

**Pull request merged**